### PR TITLE
Add Payment Method complex field type

### PR DESCRIPTION
**Pull request by chrismccluskey**
#### Changes proposed in this Pull Request

Closes 507-gh-automattic/gold

* Creates a representation of [Sift's "Payment Method" complex field type](https://developers.sift.com/docs/php/events-api/complex-field-types/payment-method).
* Adds 2 filters that can be used by authors of payment gateway plugins to extract payment method and charge data from a \WC_Order object.
* Adds additional filters to use the payment method and charge data and select properties on the Sift Payment Method type.
* Adds a basic implementation for the `woocommerce-gateway-stripe` plugin and `transact-gateway` plugin.

#### Testing instructions

* Automated tests are included.

### CODE REVIEW

**Comment by millerf** [comment]
> Adds a basic implementation for the woocommerce-gateway-stripe plugin and transact-gateway plugin.

~Is that actually working for the actual implementation of WooPayments?~ It [does](https://github.com/woocommerce/sift-for-woocommerce/pull/4/files#diff-065ed9d09683982b3369c606f75145996375707e741053b9f3888d4113874170R632) :)

File: src/inc/payment-gateway.php [hunk 1]
```diff
@@ -0,0 +1,662 @@
...
+
+	/**
+	 * Create a class to represent the payment gateway plugin used for an order.
+	 *
+	 * @param string    $woo_gateway_id The payment gateway ID as exposed by the plugin (e.g., the output of \WC_Order->get_payment_method()).
+	 * @param \WC_Order $order          The WC_Order which uses this gateway. Used to lookup additional information for payment gateway plugins which work with multiple payment gateways.
+	 */
+	public function __construct( string $woo_gateway_id, \WC_Order $order ) {
+		if ( 'transact' === strtolower( $woo_gateway_id ) ) {
+			$woo_gateway_id = 'woopayments';
```
Comment (millerf) [inline src/inc/payment-gateway.php:632]: I got my answer :)

File: src/inc/payment-gateways/paypal.php [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<?php declare( strict_types=1 );
+
+add_filter( 'sift_for_woocommerce_paypal_payment_gateway_string', fn() => '$paypal' );
```
Comment (millerf) [inline src/inc/payment-gateways/paypal.php:3]: Still not certain anyone would want to override the string for those specific payment gateways. But that is not an issue.

**Review by millerf — APPROVED** [review]
LGTM. If I am not mistaken the specifc PM properties discussed in pdjTHR-4mU-p2#comment-6424 also need a followup...

File: src/inc/payment-gateways/paypal.php (hunk 2 above)
Comment (chrismccluskey) [reply src/inc/payment-gateways/paypal.php:3]: So in this case, this filter would represent the beginning of a PayPal plugin that exposes a gateway called `paypal`. The only way that we get these values are from filters, we wouldn't guess based on the payment gateway id (prefixing it with `$` and seeing if it's valid). We could do that in the future if it's valuable.

**Pull request merged**